src/GPE.jl

```diff
@@ -351,13 +351,17 @@ function _predict(gp::GPE, x::AbstractMatrix)
     mu = mean(gp.mean, x) + cK'*gp.alpha        # Predictive mean
     Lck = whiten!(gp.cK, cK)
     Sigma_raw = cov(gp.kernel, x, x, priordata)
-    # Sigma_raw = Sigma_raw - Lck'Lck
-    LinearAlgebra.BLAS.syrk!('U', 'T', -1.0, Lck, 1.0, Sigma_raw)
-    LinearAlgebra.copytri!(Sigma_raw, 'U')
+    subtract_Lck!(eltype(x), Sigma_raw, Lck)
     # Add jitter to get stable covariance
     m, chol = make_posdef!(Sigma_raw)
     return mu, PDMat(m, chol)
 end
+@inline function subtract_Lck!(::Any, Sigma_raw, Lck)
```

Comment:
I would suggest inverting the type signatures of the two `subtract_Lck!`. `LinearAlgebra.BLAS.syrk!` only works for `AbstractMatrix{Float64}`. Your implementation of `subtract_Lck!(::Type{<:Dual}, Sigma_raw, Lck)` should be the generic one.

I'm not sure I see the need to have a first argument be the element type itself. I would just define `subtract_Lck!(Sigma_raw::AbstractMatrix, Lck::AbstractMatrix)` and `subtract_Lck!(Sigma_raw::AbstractMatrix{Float64}, Lck::AbstractMatrix{Float64})`.

Reply:
Good point, thanks.